src/jit/lower.cpp

```diff
@@ -2835,6 +2835,13 @@ void Lowering::InsertPInvokeMethodProlog()
 
     JITDUMP("======= Inserting PInvoke method prolog\n");
 
+    // If the entry block has more than one incoming edge, insert a preceding scratch BB into which we can safely
+    // insert the P/Invoke method prolog.
+    if (comp->fgFirstBB->countOfInEdges() > 1)
```

Comment:
I'd prefer the simplicity of always doing it.

Actually, I think you want to always do it and then change the code to always append to the first block (which will be guaranteed to be a scratch block). Otherwise, in the case a scratch block already exists (which might be the case already, today), the code prepends to other code that's already there. So, we're prepending before debug variable initializations, IBC counts, reverse pinvoke enter calls, just-my-code callbacks, etc.